core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java

```diff
@@ -107,10 +107,13 @@ private static boolean restore(
           LOG.error("Invalid page info");
           return false;
         }
-        metaStore.addPage(pageInfo.getPageId(), pageInfo);
+        PageId pageId = pageInfo.getPageId();
+        metaStore.addPage(pageId, pageInfo);
         if (metaStore.bytes() > pageStore.getCacheSize()) {
-          LOG.error("Loaded pages exceed cache capacity ({} bytes)", pageStore.getCacheSize());
-          return false;
+          metaStore.removePage(pageId);
+          pageStore.delete(pageId);
+          LOG.warn("Abandon {} in restore: cache size is {} bytes and {} bytes used",
+              pageInfo, pageStore.getCacheSize(), metaStore.bytes());
```

Comment:
Will this part be performant enough if there are a large number of extra pages? Is there any way to parallelize or otherwise delete the rest of the stream at once after we detect the metastore is out of space?

Reply:
removed the warning but show the aggregated result in the end.
there is no easy way to bulk-delete the files after metastore is filled